Gordon360/Services/MembershipService.cs

```diff
@@ -88,167 +92,108 @@ public MEMBERSHIP GetSpecificMembership(int membershipID)
-            return memberships.Where(m => m.GroupAdmin == true);
+            if (sessionCode == null)
+            {
+                sessionCode = this.GetCurrentSessionCode();
+            }
+            return _context.MembershipView.Where(m => m.ActivityCode == activityCode && m.SessionCode == sessionCode && m.GroupAdmin == true);
         }
 
         /// <summary>
         /// Fetches the leaders of the activity whose activity code is specified by the parameter.
         /// </summary>
         /// <param name="activityCode">The activity code.</param>
         /// <returns>MembershipViewModel IEnumerable. If no records were found, an empty IEnumerable is returned.</returns>
-        public async Task<IEnumerable<MembershipViewModel>> GetLeaderMembershipsForActivityAsync(string activityCode)
+        public IEnumerable<MembershipView> GetLeaderMembershipsForActivity(string activityCode)
         {
             var leaderRole = Helpers.GetLeaderRoleCodes();
-            var memberships = await GetMembershipsForActivityAsync(activityCode);
-            return memberships.Where(x => x.Participation == leaderRole);
+            return _context.MembershipView.Where(m => m.ActivityCode == activityCode && m.Participation == leaderRole);
         }
 
         /// <summary>
         /// Fetches the advisors of the activity whose activity code is specified by the parameter.
         /// </summary>
         /// <param name="activityCode">The activity code.</param>
         /// <returns>MembershipViewModel IEnumerable. If no records were found, an empty IEnumerable is returned.</returns>
-        public async Task<IEnumerable<MembershipViewModel>> GetAdvisorMembershipsForActivityAsync(string activityCode)
+        public IEnumerable<MembershipView> GetAdvisorMembershipsForActivity(string activityCode)
         {
 
             var advisorRole = Helpers.GetAdvisorRoleCodes();
-            var memberships = await GetMembershipsForActivityAsync(activityCode);
-            return memberships.Where(x => x.Participation == advisorRole);
+            return _context.MembershipView.Where(m => m.ActivityCode == activityCode && m.Participation == advisorRole);
         }
 
         /// <summary>
         /// Fetches all the membership information linked to the student whose id appears as a parameter.
         /// </summary>
         /// <param name="username">The student's AD Username.</param>
         /// <returns>A MembershipViewModel IEnumerable. If nothing is found, an empty IEnumerable is returned.</returns>
-        public async Task<IEnumerable<MembershipViewModel>> GetMembershipsForStudentAsync(string username)
+        public IEnumerable<MembershipView> GetMembershipsByUser(string username)
         {
-            var account = _context.ACCOUNT.FirstOrDefault(x => x.AD_Username.Trim() == username);
+            var account = _context.ACCOUNT.FirstOrDefault(a => a.AD_Username.Trim() == username);
```

Comment:
Would it be easier to use the `accountService.GetAccountByUsername` method here?